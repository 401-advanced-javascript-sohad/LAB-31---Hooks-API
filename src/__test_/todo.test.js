/* eslint-disable no-unused-vars */
import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import ToDo from '../modules/todo.js';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<ToDo />', ()=>{
  it('exists in application start',()=>{
    let app = shallow(<ToDo />);
    expect(app.find('form').exists()).toBeTruthy();
    expect(app.find('input').exists()).toBeTruthy();
    expect(app.find('button').exists()).toBeTruthy();
    
});



    it('exist in our app' , () => {
    let test = shallow(<ToDo/>);
    expect(test.find('button').exists()).toBeTruthy();
 });


  it('click simulate test',()=>{
    let app = mount(<ToDo />);
    let button = app.find('button');
    button.simulate('click');
    expect(typeof button).toBe('object');
    expect(typeof app).toBe('object');
  });


  it('correct render', () => {
    const tree = renderer.create(<ToDo />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  
    it('Assert DOM in snapshot', ()=> {
      let test = renderer.create(<ToDo />);
      expect(test).toMatchSnapshot();
    });
});






// });
