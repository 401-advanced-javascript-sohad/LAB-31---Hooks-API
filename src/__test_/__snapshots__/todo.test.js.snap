// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToDo /> Assert DOM in snapshot 1`] = `
<main>
  <img
    id="img1"
    src="https://media1.tenor.com/images/1f729b628843b6e9739c5b708e5beb9f/tenor.gif?itemid=9979305"
  />
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Type your task"
      requigreen={true}
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Difficulty (1-5)"
      requigreen={true}
      type="number"
    />
    <button
      type="submit"
    >
      Add
    </button>
  </form>
</main>
`;

exports[`<ToDo /> Assert DOM stability via snapshot 1`] = `
<main>
  <img
    id="img1"
    src="https://media1.tenor.com/images/1f729b628843b6e9739c5b708e5beb9f/tenor.gif?itemid=9979305"
  />
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Type your task"
      requigreen={true}
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Difficulty (1-5)"
      requigreen={true}
      type="number"
    />
    <button
      type="submit"
    >
      Add
    </button>
  </form>
</main>
`;

exports[`<ToDo /> correct render 1`] = `
<main>
  <img
    id="img1"
    src="https://media1.tenor.com/images/1f729b628843b6e9739c5b708e5beb9f/tenor.gif?itemid=9979305"
  />
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Type your task"
      requigreen={true}
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Difficulty (1-5)"
      requigreen={true}
      type="number"
    />
    <button
      type="submit"
    >
      Add
    </button>
  </form>
</main>
`;

exports[`<ToDo /> correcte render 1`] = `
<main>
  <img
    id="img1"
    src="https://media1.tenor.com/images/1f729b628843b6e9739c5b708e5beb9f/tenor.gif?itemid=9979305"
  />
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Type your task"
      requigreen={true}
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Difficulty (1-5)"
      requigreen={true}
      type="number"
    />
    <button
      type="submit"
    >
      Add
    </button>
  </form>
</main>
`;

exports[`<ToDo /> correctly render 1`] = `
<main>
  <img
    id="img1"
    src="https://media1.tenor.com/images/1f729b628843b6e9739c5b708e5beb9f/tenor.gif?itemid=9979305"
  />
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Type your task"
      requigreen={true}
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Difficulty (1-5)"
      requigreen={true}
      type="number"
    />
    <button
      type="submit"
    >
      Add
    </button>
  </form>
</main>
`;

exports[`<ToDo /> renders  1`] = `
<main>
  <img
    id="img1"
    src="https://media1.tenor.com/images/1f729b628843b6e9739c5b708e5beb9f/tenor.gif?itemid=9979305"
  />
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Type your task"
      requigreen={true}
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Difficulty (1-5)"
      requigreen={true}
      type="number"
    />
    <button
      type="submit"
    >
      Add
    </button>
  </form>
</main>
`;

exports[`<ToDo /> renders correctly 1`] = `
<main>
  <img
    id="img1"
    src="https://media1.tenor.com/images/1f729b628843b6e9739c5b708e5beb9f/tenor.gif?itemid=9979305"
  />
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Type your task"
      requigreen={true}
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Difficulty (1-5)"
      requigreen={true}
      type="number"
    />
    <button
      type="submit"
    >
      Add
    </button>
  </form>
</main>
`;
